TowerDefense/GameObjects/Enemy.cs

```diff
@@ -54,6 +63,21 @@ public void damage(int damage)
 
         public bool Update(GameTime gameTime)
         {
+            if (HP <= 0)
+            {
+                color.R -= 15;
+                color.G -= 15;
+                color.B -= 15;
+                if (color == Color.Black)
+                {
+                    return true;
```

Comment:
Yeah I used to have that, but this makes the update cycle and the clean up cycle the exact same line.  I wrote it for something then ended up applying the pattern everywhere.  Again not sure if it's a great pattern, but it's super succinct.